### PR TITLE
tests: Implement DomainBreakService test cases

### PR DESCRIPTION
**Pull request by playfulCloud**

### CODE REVIEW

**Comment by playfulCloud** [comment]
Those test cases should me unitilize

File: app/src/main/java/com/dreamtracker/app/view/adapters/api/BreakComponentResponse.java [hunk 1]
```diff
@@ -4,10 +4,33 @@
...
 
 @RequiredArgsConstructor
 @Builder
 @Getter
 public class BreakComponentResponse extends StatsComponentResponse {
     private final double averageBreak;
+
+
+    @Override
+    public String toString() {
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/adapters/api/BreakComponentResponse.java:18]: Just use lombok @Value or simply java record for things like this

File: app/src/main/java/com/dreamtracker/app/view/adapters/api/StatsComponentResponse.java [hunk 2]
```diff
@@ -1,3 +1,6 @@
 package com.dreamtracker.app.view.adapters.api;
 
+
+import lombok.Builder;
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/adapters/api/StatsComponentResponse.java:4]: nit: unused import

File: app/src/main/java/com/dreamtracker/app/view/domain/model/aggregate/BreaksAggregate.java [hunk 3]
```diff
@@ -23,14 +24,32 @@ public class BreaksAggregate {
     private int breaksQuantity;
     private boolean isBreak;
 
+//    @Override
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/model/aggregate/BreaksAggregate.java:27]: :(

File: app/src/main/java/com/dreamtracker/app/view/domain/model/aggregate/BreaksAggregate.java [hunk 4]
```diff
@@ -23,14 +24,32 @@ public class BreaksAggregate {
...
     @Override
-    public String toString() {
-        return "BreaksAggregate{" +
-                "id=" + id +
-                ", habitUUID=" + habitUUID +
-                ", sumOfBreaks=" + sumOfBreaks +
-                ", breaksQuantity=" + breaksQuantity +
-                ", isBreak=" + isBreak +
-                '}';
+    public boolean equals(Object o) {
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/model/aggregate/BreaksAggregate.java:39]: Use lombok @Data or @EqualsAndHashcode for this

File: app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java [hunk 5]
```diff
@@ -28,13 +27,14 @@ public StatsComponentResponse initializeAggregates(UUID habitId) {
...
       UUID habitId, HabitTrackResponse habitTrackResponse) {
-    var aggregateFoundByHabitId =
+    BreaksAggregate aggregateFoundByHabitId =
         breaksAggregateRepositoryPort
             .findByHabitUUID(habitId)
             .orElseThrow(
                 () ->
                     new EntityNotFoundException(ExceptionMessages.entityNotFoundExceptionMessage));
 
+    System.out.println(aggregateFoundByHabitId);
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java:37]: Use a proper logger instead like slf4j facade + any engine

File: app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java [hunk 6]
```diff
@@ -65,12 +65,13 @@ public StatsComponentResponse getCalculateResponse(UUID habitId) {
   }
 
   private StatsComponentResponse mapToResponse(BreaksAggregate breaksAggregate) {
-    return BreakComponentResponse.builder()
-        .averageBreak(calculateAverageBreak(breaksAggregate))
-        .build();
+    return BreakComponentResponse.builder().averageBreak(calculateAverageBreak(breaksAggregate)).build();
   }
 
   private double calculateAverageBreak(BreaksAggregate breaksAggregate) {
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java:71]: I advise to use BigDecimal for calculations or some kind of microtype to be able to have the same rounding, validation etc

File: app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java [hunk 7]
```diff
@@ -65,12 +65,13 @@ public StatsComponentResponse getCalculateResponse(UUID habitId) {
...
+    if(breaksAggregate.getBreaksQuantity() == 0){
+      return 0;
+    }
     return (double) breaksAggregate.getSumOfBreaks() /breaksAggregate.getBreaksQuantity();
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java:75]: could be a method in microtype

File: app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java [hunk 8]
```diff
@@ -65,12 +65,13 @@ public StatsComponentResponse getCalculateResponse(UUID habitId) {
...
 
   private StatsComponentResponse mapToResponse(BreaksAggregate breaksAggregate) {
-    return BreakComponentResponse.builder()
-        .averageBreak(calculateAverageBreak(breaksAggregate))
-        .build();
+    return BreakComponentResponse.builder().averageBreak(calculateAverageBreak(breaksAggregate)).build();
   }
 
   private double calculateAverageBreak(BreaksAggregate breaksAggregate) {
+    if(breaksAggregate.getBreaksQuantity() == 0){
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainBreaksService.java:72]: could be a method in microtype

File: app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainStreakService.java [hunk 9]
```diff
@@ -20,6 +20,7 @@ public class DomainStreakService implements StatsTemplate {
   @Override
   public StatsComponentResponse initializeAggregates(UUID habitId) {
     var streakAggregate = initialize(habitId);
+    System.out.println(streakAggregate);
```
Comment (pawel-ochrymowicz) [inline app/src/main/java/com/dreamtracker/app/view/domain/ports/statistics/DomainStreakService.java:23]: please proper logger

File: app/src/test/java/com/dreamtracker/app/view/domain/model/aggregateManagers/BreakAggregateManagerTest.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package com.dreamtracker.app.view.domain.model.aggregateManagers;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+class BreakAggregateManagerTest {
```
Comment (pawel-ochrymowicz) [inline app/src/test/java/com/dreamtracker/app/view/domain/model/aggregateManagers/BreakAggregateManagerTest.java:6]: :(

File: app/src/test/java/com/dreamtracker/app/view/domain/model/aggregateManagers/StatsAggregatorTest.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package com.dreamtracker.app.view.domain.model.aggregateManagers;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+class StatsAggregatorTest {
```
Comment (pawel-ochrymowicz) [inline app/src/test/java/com/dreamtracker/app/view/domain/model/aggregateManagers/StatsAggregatorTest.java:7]: Please try TDD, so your classes will be more testable and you will never commit code without tests. IMO, TDD makes you writing better designed code.

Comment (playfulCloud) [reply app/src/test/java/com/dreamtracker/app/view/domain/model/aggregateManagers/StatsAggregatorTest.java:7]: oh yeah sorry i forgot to clean all the drafts after implementation

**Pull request merged**